`expiring-todo-comments` should include the comment itself in the report

Copied from https://github.com/eslint/eslint/issues/12327

---

I think the point of this rule is to keep track of TODO comments, however this is hard to follow; I'd have to open each file separately to see if I can handle a task.

# Current

```ts
  source/globals.d.ts:1:1
  ⚠    1:1   Unexpected todo comment.                                                                   no-warning-comments
  ⚠    2:1   Unexpected todo comment.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   Unexpected todo comment.                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   Unexpected todo comment.                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   Unexpected todo comment.                                                                   no-warning-comments
```

# Ideal


```ts
  source/globals.d.ts:1:1
  ⚠    1:1   Unexpected todo comment: Drop ignore after #9.                                                                  no-warning-comments
  ⚠    2:1   Unexpected todo comment: Find this info on the page.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   Unexpected todo comment: Improve JSX types for event listeners so we can use... [clipped]                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   Unexpected todo comment: Drop some definitions when their related bugs are resolved                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   Unexpected todo comment: Drop `as` after https://github.com/microsoft/TSJS-lib-generator/issues/741                                                                   no-warning-comments
```

Or


```ts
  source/globals.d.ts:1:1
  ⚠    1:1   TODO: Drop ignore after #9.                                                                  no-warning-comments
  ⚠    2:1   FIXME: Find this info on the page.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   TODO: Improve JSX types for event listeners so we can use... [clipped]                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   TODO: Drop some definitions when their related bugs are resolved                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   TODO: Drop `as` after https://github.com/microsoft/TSJS-lib-generator/issues/741                                                                   no-warning-comments
```

> // @lubien In case you're interested in working on this.

I can work on this weekend, sure!



> I would go for the short version:

Noted.

How long until clip? I could use something like `process.stdout.columns` to calculate the width and try not to overflow too much. I'd have also to count ESLint own input.

```sh
lubien@heron:~/dev$ npx eslint Untitled-1.js

~/dev/foo.js
  1:1  error  Unexpected 'todo' comment                                    unicorn/expiring-todo-comments

✖ 1 problem (1 errors, 0 warnings)

lubien@heron:~/dev$ node
> process.stdout.columns
103
```

Given 103 column digits. For this case I can see 4 digits for reporting level, a few for `line:column` (in this case just 3, but would need more). Double space separators at first then some spaces to make the rule name appear at right. 61 digits for the report message.

How about making messages at most 50% `process.stdout.columns`? With a fallback width in case some situation make this value not available (happened to me at least in Elixir I don't know about NodeJS).

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
 		"eslint-template-visitor": "^2.2.1",
 		"eslint-utils": "^2.1.0",
 		"import-modules": "^2.0.0",
-		"lodash": "^4.17.15",
+		"lodash": "^4.17.20",
 		"pluralize": "^8.0.0",
 		"read-pkg-up": "^7.0.1",
 		"regexp-tree": "^0.1.21",
@@ -49,26 +49,26 @@
 	},
 	"devDependencies": {
 		"@ava/babel": "^1.0.1",
-		"@babel/code-frame": "7.10.3",
+		"@babel/code-frame": "7.10.4",
 		"@lubien/fixture-beta-package": "^1.0.0-beta.1",
-		"@typescript-eslint/parser": "^3.4.0",
-		"ava": "^3.9.0",
+		"@typescript-eslint/parser": "^4.0.1",
+		"ava": "^3.12.1",
 		"babel-eslint": "^10.1.0",
 		"chalk": "^4.1.0",
-		"eslint": "^7.3.0",
+		"eslint": "^7.8.0",
 		"eslint-ava-rule-tester": "^4.0.0",
 		"eslint-plugin-eslint-plugin": "^2.3.0",
-		"execa": "^4.0.2",
+		"execa": "^4.0.3",
 		"listr": "^0.14.3",
 		"nyc": "^15.1.0",
 		"outdent": "^0.7.1",
 		"pify": "^5.0.0",
-		"typescript": "^3.9.5",
+		"typescript": "^4.0.2",
 		"vue-eslint-parser": "^7.1.0",
-		"xo": "^0.32.0"
+		"xo": "^0.33.1"
 	},
 	"peerDependencies": {
-		"eslint": ">=7.3.0"
+		"eslint": ">=7.8.0"
 	},
 	"ava": {
 		"babel": true,

--- a/test/expiring-todo-comments.js
+++ b/test/expiring-todo-comments.js
@@ -48,8 +48,8 @@ const missingAtSymbolError = (bad, good, message) => ({
 	message: `Missing '@' on TODO argument. On '${bad}' use '${good}'. ${message}`
 });
 
-const noWarningCommentError = () => ({
-	message: 'Unexpected \'todo\' comment.'
+const noWarningCommentError = comment => ({
+	message: `Unexpected 'todo' comment: '${comment}'.`
 });
 
 ruleTester.run('expiring-todo-comments', rule, {
@@ -277,37 +277,37 @@ ruleTester.run('expiring-todo-comments', rule, {
 		},
 		{
 			code: '// TODO',
-			errors: [noWarningCommentError()],
+			errors: [noWarningCommentError('TODO')],
 			options: [{allowWarningComments: false}]
 		},
 		{
 			code: '// TODO []',
-			errors: [noWarningCommentError()],
+			errors: [noWarningCommentError('TODO []')],
 			options: [{allowWarningComments: false}]
 		},
 		{
 			code: '// TODO [no meaning at all]',
-			errors: [noWarningCommentError()],
+			errors: [noWarningCommentError('TODO [no meaning at all]')],
 			options: [{allowWarningComments: false}]
 		},
 		{
 			code: '// TODO [] might have [some] that [try [to trick] me]',
-			errors: [noWarningCommentError()],
+			errors: [noWarningCommentError('TODO [] might have [some] that [try [to...')],
 			options: [{allowWarningComments: false}]
 		},
 		{
 			code: '// TODO [but [it will]] [fallback] [[[ to the default ]]] rule [[[',
-			errors: [noWarningCommentError()],
+			errors: [noWarningCommentError('TODO [but [it will]] [fallback] [[[ to...')],
 			options: [{allowWarningComments: false}]
 		},
 		{
 			code: '// TODO [engine:npm@>=10000]: Unsupported engine',
-			errors: [noWarningCommentError()],
+			errors: [noWarningCommentError('TODO [engine:npm@>=10000]: Unsupported...')],
 			options: [{allowWarningComments: false}]
 		},
 		{
 			code: '// TODO [engine:somethingrandom@>=10000]: Unsupported engine',
-			errors: [noWarningCommentError()],
+			errors: [noWarningCommentError('TODO [engine:somethingrandom@>=10000]:...')],
 			options: [{allowWarningComments: false}]
 		},
 		{
@@ -371,7 +371,7 @@ ruleTester.run('expiring-todo-comments', rule, {
 			code: '// TODO [ISSUE-123] fix later',
 			options: [{allowWarningComments: false, ignore: []}],
 			errors: [
-				noWarningCommentError()
+				noWarningCommentError('TODO [ISSUE-123] fix later')
 			]
 		},
 		{
@@ -381,7 +381,7 @@ ruleTester.run('expiring-todo-comments', rule, {
 			`,
 			options: [{allowWarningComments: false, ignore: [/issue-\d+/i]}],
 			errors: [
-				noWarningCommentError()
+				noWarningCommentError('TODO fix later')
 			]
 		},
 		{
@@ -391,7 +391,7 @@ ruleTester.run('expiring-todo-comments', rule, {
 			*/`,
 			options: [{allowWarningComments: false, ignore: [/issue-\d+/i]}],
 			errors: [
-				noWarningCommentError()
+				noWarningCommentError('TODO Invalid')
 			]
 		}
 	]

--- a/test/integration/projects.js
+++ b/test/integration/projects.js
@@ -72,11 +72,10 @@ module.exports = [
 			'aio/content/examples/router/src/app/app-routing.module.9.ts'
 		]
 	},
-	// TODO: enable this when ``@typescript-eslint/parser` support typescript 4.0
-	// {
-	// 	repository: 'https://github.com/microsoft/typescript',
-	// 	extraArguments: typescriptArguments
-	// },
+	{
+		repository: 'https://github.com/microsoft/typescript',
+		extraArguments: typescriptArguments
+	},
 	{
 		repository: 'https://github.com/microsoft/vscode',
 		extraArguments: typescriptArguments

